### PR TITLE
Integration specs for views

### PR DESCRIPTION
**Pull request by cynthiainga**
### Project requirements
- Use Capybara to write integration tests for each view in your project.
- Login page:
  - I can see the username and password inputs and the "Submit" button.
  - When I click the submit button without filling in the username and the password, I get a detailed error.
  - When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
  - When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.
- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.
- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.
 
To run the test you need to run the following command in your terminal `bundle exec rspec spec/views`

### CODE REVIEW

File: spec/views/users/login.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+RSpec.describe 'login page', type: :system do
+  describe 'login page' do
+    it 'I can see the username and password inputs and the Submit button.' do
+      visit new_user_session_path
+      expect(page).to have_field('user_email')
+      expect(page).to have_field('user_password')
+      expect(page).to have_button(type: 'submit')
+    end
+  end
+end
```
Comment (TedLivist) [inline spec/views/users/login.html.erb_spec.rb]: Kindly add more tests that cover the following scenarios.

- [ ] Display error when no email AND password is entered and the submit button is clicked
- [ ] Display error when an incorrect email OR password is entered and the submit button is clicked.
- [ ] Redirect to root page when correct email and password are entered and submit button is clicked.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @cynthiainga and @Thermiee,

Good job so far!
✔️ Good documentation
✔️ Correct use of capybara for testing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/users/login.html.erb_spec.rb (hunk 1 above)
Comment (cynthiainga) [reply spec/views/users/login.html.erb_spec.rb]: Hello @TedLivist , yeah sure l already implemented it, thanks for the review.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @cynthiainga, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- You have done a great job working on the requested changes ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cynthiainga** [comment]
Thanks @elisha2kyakpo1 :+1:

**Pull request merged**